flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/delegation/BatchPlanner.scala

```diff
@@ -169,4 +172,15 @@ class BatchPlanner(
   override def explainJsonPlan(jsonPlan: String, extraDetails: ExplainDetail*): String = {
     throw new TableException("This method is not supported for batch planner now.")
   }
+
+  override def isSpecifiedPlanner(): Unit = {
+    super.isSpecifiedPlanner();
+    if (!config.getConfiguration.get(ExecutionOptions.RUNTIME_MODE)
+      .equals(RuntimeExecutionMode.BATCH)) {
+      throw new IllegalArgumentException("Expect STREAMING mode but get BATCH mode. " +
```

Comment:
Rename to: `Mismatch between configured planner and actual planner. Currently, the 'execution.runtime-mode' can only be set when instantiating the table environment. Subsequent changes are not supported. Please instatiate a new TableEnvironment if necessary.`
`

Reply:
Also make it a `TableException` and update the other locations.